meter/shelly.go

```diff
@@ -1,33 +1,100 @@
 package meter
 
 import (
+	"math"
+	"strings"
+	"time"
+
 	"github.com/evcc-io/evcc/api"
 	"github.com/evcc-io/evcc/meter/shelly"
 	"github.com/evcc-io/evcc/util"
 )
 
+// Shelly meter considering usage
+type Shelly struct {
+	conn  *shelly.Connection
+	usage string
+}
+
 // Shelly meter implementation
 func init() {
 	registry.Add("shelly", NewShellyFromConfig)
 }
 
+//go:generate go tool decorate -f decorateShelly -b *Shelly -r api.Meter -t "api.PhaseVoltages,Voltages,func() (float64, float64, float64, error)" -t "api.PhaseCurrents,Currents,func() (float64, float64, float64, error)" -t "api.PhasePowers,Powers,func() (float64, float64, float64, error)"
```

Comment:
Ich frag nochmal: wozu die ganze Magie?

Reply:
Ich muss in CurrentPowers ja die Usage berücksichtigen, um bei PV den absoluten Wert zu nehmen (also imm positiv.
Dachte das geht nicht anders ...